universal_transfer_operator/src/universal_transfer_operator/data_providers/base.py

```diff
@@ -57,11 +67,11 @@ def check_if_transfer_supported(self, source_dataset: DatasetType) -> bool:
         source_connection_type = get_dataset_connection_type(source_dataset)
         return Location(source_connection_type) in self.transfer_mapping
 
-    def read(self):
+    def read(self) -> Iterator[pd.DataFrame] | Iterator[FileStream]:
         """Read the dataset and write to local reference location"""
         raise NotImplementedError
 
-    def write(self, source_ref):
+    def write(self, source_ref) -> str:  # type: ignore
         """Write the data from local reference location to the dataset"""
```

Comment:
Same here, are we moving a file located in the local file system to a target destination (file, table..?)

Reply:
Done.